File: solutions/0861. Score After Flipping Matrix/0861_2.py
# Time Complexity: O(mn)
# Space Complexity: O(1)
class Solution:
  def matrixScore(self, grid: List[List[int]]) -> int:
    m = len(grid)
    n = len(grid[0])
    ans = m

    for j in range(1, n):
      onesCount = sum(grid[i][j] == grid[i][0] for i in range(m))
      ans = ans * 2 + max(onesCount, m - onesCount)

    return ans

File: solutions/0861. Score After Flipping Matrix/0861_2.rs
// Time Complexity: O(mn)
// Space Complexity: O(1)
impl Solution {
    pub fn matrix_score(grid: Vec<Vec<i32>>) -> i32 {
        let m = grid.len();
        let n = grid[0].len();
        let mut ans = m as i32;

        for j in 1..n {
            let ones_count = grid
                .iter()
                .map(|row| (row[j] == row[0]) as i32)
                .sum::<i32>();
            ans = ans * 2 + std::cmp::max(ones_count, m as i32 - ones_count);
        }

        ans
    }
}
